pipeline/src/pipeline.rs

```diff
@@ -677,6 +706,18 @@ impl<T: FieldElement> Pipeline<T> {
         Ok(())
     }
 
+    fn maybe_write_serialized_pil(
+        &self,
+        pil: &Analyzed<T>,
+        suffix: &str,
+    ) -> Result<(), Vec<String>> {
+        if let Some(path) = self.path_if_should_write(|name| format!("{name}{suffix}.cbor"))? {
```

Comment:
Beyond the change of file extension,
we might want to add a magic header and a version if we ever want forward compatibility or different PIL compilers like Polygon to add a loader for our own PIL.

Reply:
Agree, though I think we can do it in a different PR.